### PR TITLE
Fixes #25265 - Allow I18n in defaultProps

### PR DESCRIPTION
**Pull request by sharvit**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteSearchButton.js [hunk 1]
```diff
@@ -3,22 +3,24 @@ import PropTypes from 'prop-types';
...
 );
 
 SearchButton.propTypes = {
   className: PropTypes.string,
+  children: PropTypes.node,
 };
 
 SearchButton.defaultProps = {
   className: '',
+  children: __('Search'),
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteSearchButton.js:23]: Basically, this is what this PR allows us to do :)

**Review by ohadlevy — APPROVED** [review]
looks good, will wait for github to come back online to start the tests.

**Review by amirfefer — COMMENTED** [review]
Thanks @sharvit :+1: 
it seems that it won't work on plugins
is there an alternative for plugins, rather then copy paste the entire mocks ?

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteSearchButton.js (hunk 1 above)
Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteSearchButton.js:23]: how does moving the mocks from `test_helpers` impacts on PropTypes?

**Comment by sharvit** [comment]
@amirfefer @ohadlevy I think the best solution is foreman as NPM package (loaded from github instead actually npm package).

**Comment by ohadlevy** [comment]
> @amirfefer @ohadlevy I think the best solution is foreman as NPM package (loaded from github instead actually npm package).

how about doing a quick POC? :)

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteSearchButton.js (hunk 1 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteSearchButton.js:23]: I didn't really get it but as far as it weird, it seems the `test_setup.js` running after the component file gets loaded.

**Comment by sharvit** [comment]
Yes I’m going to try it soon!

**Comment by ohadlevy** [comment]
@amirfefer I guess this can still be merged in its current form regardless?

**Review by amirfefer — APPROVED** [review]
LGTM !
thanks @sharvit  :+1:

**Pull request merged**